### PR TITLE
Removing ThinLTO support

### LINKED ISSUE #6064: Thin LTO doesn't work
Hi crystal community!

I tried to use `--lto` flag by using  `--lto=thin`:

```
--lto=FLAG                       Use ThinLTO --lto=thin
```

Then I compile my code using:

```
crystal build src/scry.cr --stats --release --lto=thin
```

and I get this error:

```
Invalid memory access (signal 11) at address 0x8
[0x5555d12e9fb6] ???
[0x5555d0b086fa] __crystal_sigfault_handler +42
[0x7f6a80685dd0] ???
[0x7f6a82c60c20] ???
[0x7f6a82c60cc4] _ZN4llvm18ProfileSummaryInfo15getProfileCountEPKNS_11InstructionEPNS_18BlockFrequencyInfoE +52
[0x7f6a82c502b5] ???
[0x7f6a82c51f4e] _ZN4llvm23buildModuleSummaryIndexERKNS_6ModuleESt8functionIFPNS_18BlockFrequencyInfoERKNS_8FunctionEEEPNS_18ProfileSummaryInfoE +702
[0x5555d12ed73c] LLVMWriteBitcodeWithSummaryToFile +172
[0x5555d0fd7792] ???
[0x5555d0fdbc43] ???
[0x5555d0af1b44] ???
[0x0] ???
Codegen (bc+obj):                  00:00:00.119673414 ( 252.99MB)
_main.o: file not recognized: File truncated
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
Error: execution of command failed with code: 1: `clang -flto=thin -Wl,-plugin-opt,jobs=8,-plugin-opt,cache-dir=/home/main/.cache/crystal/home-main-Projects-scry-src-scry.cr/lto.cache -O2 "${@}" -o '/home/main/Projects/scry/bin/scry'  -rdynamic  -lpcre -lm -lgc -lpthread /usr/lib/crystal/ext/libcrystal.a -levent -lrt -ldl -L/usr/lib -L/usr/local/lib`
```

See [Full logs here](https://gist.github.com/faustinoaq/f4324ee9fa545e9738f1060e64c74fe5)

### PR DESCRIPTION
**Pull request by beta-ziliani**
Solves #11193 and closes #6064.

### CODE REVIEW

File: src/compiler/crystal/compiler.cr [hunk 1]
```diff
@@ -366,19 +363,8 @@ module Crystal
...
-          {% if flag?(:darwin) %}
-            cc = ENV["CC"]? || "#{clang} -flto=thin -Wl,-mllvm,-threads=#{n_threads},-cache_path_lto,#{lto_cache_dir},#{@release ? "-mllvm,-O2" : "-mllvm,-O0"}"
-          {% else %}
-            cc = ENV["CC"]? || "#{clang} -flto=thin -Wl,-plugin-opt,jobs=#{n_threads},-plugin-opt,cache-dir=#{lto_cache_dir} #{@release ? "-O2" : "-O0"}"
-          {% end %}
-        else
-          cc = CC
-        end
-
+        cc = CC
```
Comment (wontruefree) [inline src/compiler/crystal/compiler.cr:366]: this is set up here so does this line need to be here?
https://github.com/crystal-lang/crystal/blob/b3807a67c965c3e18deeb3b09753500a7296d061/src/compiler/crystal/compiler.cr#L24

**Comment by straight-shoota** [comment]
Such a hard removal from the CLI isn't ideal. I'd suggest to keep responding to the `--lto` option, but don't accept any value i.e. error unconditionally. 

I doubt this is used mutch, but if it is, users will receive a proper error message instead of just `invalid option: --lto`.

**Comment by beta-ziliani** [comment]
> Such a hard removal from the CLI isn't ideal. 

Normally I would agree with that, but AFAIK it isn't working since v0.25.0, so it's not something people could be using anyway.

**Review by sdogruyol — APPROVED** [review]
Thank you @beta-ziliani 🙏

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**